packages/lesswrong/components/hooks/useOnNavigate.tsx

```diff
@@ -38,11 +42,11 @@ const NavigationEventSender = () => {
  * at the *start* of the navigation, ie, when the link is first clicked (but
  * before most of the stuff at the destination has loaded).
  */
-export function useOnNavigate(fn: ({oldLocation,newLocation}: {oldLocation: RouterLocation|null, newLocation: RouterLocation})=>void) {
+export function useOnNavigate(fn: (change: LocationChange)=>void) {
   useEffect(() => {
-    routerOnUpdate.add(fn);
+    onNavigateFunctions.push(fn);
     return () => {
-      routerOnUpdate.remove(fn);
+      onNavigateFunctions = onNavigateFunctions.filter(f=>f!==fn);
```

Comment:
Seems like this should be fine but I always feel a bit uneasy about performing reference-equality comparisons.

Reply:
It's fine because the reference added to the array and the reference removed from the array later are coming from the same variable.